modules/Microsoft.Cache/redis/deploy.bicep

```
@@ -0,0 +1,262 @@
+@description('Optional. The location to deploy the Redis cache service.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Redis cache resource.')
+param name string
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Enables system assigned managed identity on the resource.')
+param systemAssignedIdentity bool = false
+
+@description('Optional. The ID(s) to assign to the resource.')
+param userAssignedIdentities object = {}
+
+@description('Optional. Specifies whether the non-ssl Redis server port (6379) is enabled.')
+param enableNonSslPort bool = false
+
+@allowed([
+  '1.0'
+  '1.1'
+  '1.2'
+])
+@description('Optional. Specify the type of lock.')
+param minimumTlsVersion string = '1.2'
+
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled.')
+param publicNetworkAccess string = 'Enabled'
```

I've opened a new issue #1660 (based on issue #819) to align this to all modules as the implementation is still quite inconsistent. Suggested to align to what done for cognitive services:
```suggestion
@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set.')
@allowed([
  ''
  'Enabled'
  'Disabled'
])
param publicNetworkAccess string = ''
```

---

Makes sense for me. The only thing is that this has to be tested granularly, as I experienced multiple issue with RPs, handling null/empty values differently.